### ISSUE TITLE
An inquiry about the reproduction of your paper

### ISSUE THREAD

**Issue by JaspinXu** [abhrac/trd#2]
The work was so interesting and it was attractive to me. I'm going to follow your work recently, I reproduced your paper using your official implementation available on github, but it turns out that it does not achieve 92% accuracy for the CUB dataset, I would like to ask if there is any specific parameter or setup that can be used to achieve this sota result. I promise they will be used only for research purposed.

**Comment by abhrac** [comment]
Thank you for your interest in your work. I had recently done some refactoring to the internal organization of the repo which could have broken some of the behaviour leading to the issue that you are facing. I am trying to reproduce and fix this in my own setup. Currently, the most stable pipeline is that of proxy_graph. So, can you please try using the following command?
`
python3 src/main.py --dataset='CUB' --seed=0 --model-type=proxy_graph --train-backbone --crop-mode=random --local-weight=1 --train-bsize=16 --gpu=1 --backbone-type=disjoint_encoder
`
I have updated the documentation to reflect this.

**Comment by abhrac** [comment]
I think I have been able to figure out the cause for the issue. TRD was trained using the backbone of Relational Proxies [1], which was further trained on top of the backbone from MMAL [2]. Because of these transitive dependencies, running the training script from scratch won't provide the expected results.

So, I have uploaded the pretrained weights for CUB which can be used to reproduce the results (I have been able to reproduce the 92% accuracy on CUB on our system).
First download the model for CUB from [here](https://drive.google.com/drive/folders/1L79fXc8MnvnLA1tOzOlURItyg2QT9hdu?usp=sharing) and place it under the folder ./checkpoint/CUB/, where ./checkpoint is under the project root.
After this, run the following command for evaluation:
`
python3 src/main.py --dataset='CUB' --seed=0 --model-type=proxy_graph --train-backbone --crop-mode=random --local-weight=1 --train-bsize=8 --gpu=1 --backbone-type=disjoint_encoder --pretrained --eval_only
`

Apologies for not clarifying this in the documentation, which I have now updated with the above instructions. Hope this helps resolve your issue.

References:

[1] Abhra Chaudhuri, Massimiliano Mancini, Zeynep Akata, and Anjan Dutta. Relational proxies: Emergent relationships as fine-grained discriminators. In NeurIPS, 2022.

[2] Fan Zhang, Meng Li, Guisheng Zhai, and Yizhao Liu. Multi-branch and Multi-scale Attention Learning for Fine-Grained Visual Categorization. In MMM, 2021.

**Comment by JaspinXu** [comment]
> I think I have been able to figure out the cause for the issue. TRD was trained using the backbone of Relational Proxies [1], which was further trained on top of the backbone from MMAL [2]. Because of these transitive dependencies, running the training script from scratch won't provide the expected results.
> 
> So, I have uploaded the pretrained weights for CUB which can be used to reproduce the results (I have been able to reproduce the 92% accuracy on CUB on our system). First download the model for CUB from [here](https://drive.google.com/drive/folders/1L79fXc8MnvnLA1tOzOlURItyg2QT9hdu?usp=sharing) and place it under the folder ./checkpoint/CUB/, where ./checkpoint is under the project root. After this, run the following command for evaluation: `python3 src/main.py --dataset='CUB' --seed=0 --model-type=proxy_graph --train-backbone --crop-mode=random --local-weight=1 --train-bsize=8 --gpu=1 --backbone-type=disjoint_encoder --pretrained --eval_only`
> 
> Apologies for not clarifying this in the documentation, which I have now updated with the above instructions. Hope this helps resolve your issue.
> 
> References:
> 
> [1] Abhra Chaudhuri, Massimiliano Mancini, Zeynep Akata, and Anjan Dutta. Relational proxies: Emergent relationships as fine-grained discriminators. In NeurIPS, 2022.
> 
> [2] Fan Zhang, Meng Li, Guisheng Zhai, and Yizhao Liu. Multi-branch and Multi-scale Attention Learning for Fine-Grained Visual Categorization. In MMM, 2021.

Thank you for taking the time to solve my problem, following the pre-training file and test instructions you provided does achieve 92.45% accuracy, thank you!
